### ISSUE TITLE
ACI_REST module not working with ACI version 1.2(3c)

### ISSUE THREAD

**Issue by rdsmf** [jedelman8/aci-ansible#1]
I ran my Ansible playbook using the ACI_REST module and I am getting this error.....

TASK [aci_rest] ****************************************************************
task path: /etc/ansible/playbook/ACI-REST-TEST.yml:9
fatal: [192.168.250.50]: FAILED! => {"failed": true, "msg": "module (aci_rest) is missing interpreter line"}

Here is my playbook.......

---

- name: GET NXOS DIR
  hosts: 192.168.250.50
  gather_facts: no

  tasks:

  - aci_rest: action=get uri=/api/node/class/fvTenant.json protocol=https host={{ inventory_hostname }} username=XXXXX password=XXXXX

and HOST file .........

[aci-apic]
192.168.250.50
ansible_connection=local
ansible_user=XXXXX
ansible_ssh_pass=XXXXX

**Comment by strattner** [comment]
I'm successfully running an aci-rest task (see below) against ACI 2.0(1q), but I don't think the fabric version matters with the error you are seeing - it's complaining that something is wrong with the module itself.

Did you happen to transfer the aci-ansible modules to the host you are running on, as opposed to downloading them directly to the host via a _git clone_? I'm wondering if the module file got corrupted in some way. When I look up the "missing interpreter" message I found this:

https://groups.google.com/forum/#!topic/ansible-devel/xgyfOdAUIrQ

Which was resolved by removing an extraneous character that Windows added, that had caused ansible to not parse the #!/.../python line at the beginning of the file.

This runs without error (obviously my hosts file has an 'ACI' group, and I have the appropriate variables defined):

$ cat test-aci-rest.yml 
---

 - name: Test aci-rest
   hosts: ACI
   connection: local
   gather_facts: no

   tasks:
      - name: Get Tenant list
        aci_rest: action=get uri=/api/node/class/fvTenant.json protocol=https host={{ inventory_hostname }} username={{ user }} password={{ pass }}
        register: tenants

      - debug: var=tenants

**Comment by jedelman8** [comment]
yes, sorry I missed this one @rdsmf , but thanks @strattner for helping out....  

@rdsmf did you ever get it to work?

**Comment by rdsmf** [comment]
Thanks guys for responding back guys, I will try it in my lab tomorrow. I was getting tired of doing curl commands to my APIC controller..lol! 

    On Tuesday, October 4, 2016 3:16 PM, strattner <notifications@github.com> wrote:
 

 I'm successfully running an aci-rest task (see below) against ACI 2.0(1q), but I don't think the fabric version matters with the error you are seeing - it's complaining that something is wrong with the module itself.Did you happen to transfer the aci-ansible modules to the host you are running on, as opposed to downloading them directly to the host via a git clone? I'm wondering if the module file got corrupted in some way. When I look up the "missing interpreter" message I found this:https://groups.google.com/forum/#!topic/ansible-devel/xgyfOdAUIrQWhich was resolved by removing an extraneous character that Windows added, that had caused ansible to not parse the #!/.../python line at the beginning of the file.This runs without error (obviously my hosts file has an 'ACI' group, and I have the appropriate variables defined):
$ cat test-aci-rest.yml 
   
   - name: Test aci-rest   
hosts: ACI   
connection: local   
gather_facts: notasks:      
      - name: Get Tenant list      
aci_rest: action=get uri=/api/node/class/fvTenant.json protocol=https host={{ inventory_hostname }} username={{ user }} password={{ pass }}      
register: tenants
      - debug: var=tenants

**Comment by rdsmf** [comment]
Well one thing to add is I am doing this from Linux as opposed to windows, also I just looked at the requirements for this module and it says Cobra SDK is needed, which I did not install on this linux distro with Ansible. Is the Cobra SDK something that has to be installed for this to work? 

    On Tuesday, October 4, 2016 3:16 PM, strattner <notifications@github.com> wrote:
 

 I'm successfully running an aci-rest task (see below) against ACI 2.0(1q), but I don't think the fabric version matters with the error you are seeing - it's complaining that something is wrong with the module itself.Did you happen to transfer the aci-ansible modules to the host you are running on, as opposed to downloading them directly to the host via a git clone? I'm wondering if the module file got corrupted in some way. When I look up the "missing interpreter" message I found this:https://groups.google.com/forum/#!topic/ansible-devel/xgyfOdAUIrQWhich was resolved by removing an extraneous character that Windows added, that had caused ansible to not parse the #!/.../python line at the beginning of the file.This runs without error (obviously my hosts file has an 'ACI' group, and I have the appropriate variables defined):
$ cat test-aci-rest.yml 
   
   - name: Test aci-rest   
hosts: ACI   
connection: local   
gather_facts: notasks:      
      - name: Get Tenant list      
aci_rest: action=get uri=/api/node/class/fvTenant.json protocol=https host={{ inventory_hostname }} username={{ user }} password={{ pass }}      
register: tenants
      - debug: var=tenants

**Comment by jedelman8** [comment]
You DEFINITELY need Cobra....Cobra is the Python SDK for ACI from Cisco...it's usually hosted on the APIC controller at  https://<apic-ip>/_downloads/cobra  - for exact location I'd check latest docs...it may even be downloadable from cisco.com - search for cobra egg files.

**Comment by rdsmf** [comment]
Yep I just saw the requirements and I am actually installing the SDK now from the two egg files I downloaded from the APIC. I guess that is what I get for not RTM (reading the manual)...LOL. Thanks for the response back on this. 

    On Wednesday, October 5, 2016 5:57 PM, Jason Edelman <notifications@github.com> wrote:
 

 You DEFINITELY need Cobra....Cobra is the Python SDK for ACI from Cisco...it's usually hosted on the APIC controller at https:///_downloads/cobra - for exact location I'd check latest docs...it may even be downloadable from cisco.com - search for cobra egg files.—
